src/Shared/CoreCLRAssemblyLoader.cs

```diff
@@ -12,35 +12,16 @@ namespace Microsoft.Build.Shared
     /// <summary>
     /// CoreCLR-compatible wrapper for loading task assemblies.
     /// </summary>
-    /// <remarks>
-    /// This type defines a new <see cref="AssemblyLoadContext"/>. This isn't really what we want to
-    /// do, as we aren't trying to isolate our assembly loads from the default context. All we want
-    /// to do is add support for loading task assemblies and their dependencies, which we assume are
-    /// located immediately next to them in the file system. Ideally we wouldn't derive from
-    /// <see cref="AssemblyLoadContext"/> at all, but simply hook the <see cref="AssemblyLoadContext.Resolving"/>
-    /// event to handle finding the dependencies. For the moment, however, that approach is blocked
-    /// by https://github.com/dotnet/coreclr/issues/5837.
-    /// </remarks>
-    internal sealed class CoreClrAssemblyLoader : AssemblyLoadContext
+    internal sealed class CoreClrAssemblyLoader
     {
         private readonly Dictionary<string, Assembly> _pathsToAssemblies = new Dictionary<string, Assembly>(StringComparer.OrdinalIgnoreCase);
         private readonly Dictionary<string, Assembly> _namesToAssemblies = new Dictionary<string, Assembly>();
-        private readonly List<string> _dependencyPaths = new List<string>();
+        private readonly HashSet<string> _dependencyPaths = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
         private readonly object _guard = new object();
 
-        private static readonly string[] _extensions = new[] { "ni.dll", "ni.exe", "dll", "exe" };
+        private bool _resolvingHandlerHookedUp = false;
 
-        /// <summary>
-        /// Creates a new instance of <see cref="CoreClrAssemblyLoader" />,
-        /// sets that instance to be the default <see cref="AssemblyLoadContext" />,
-        /// and returns that instance. Throws if the Default is already set or the
-        /// binding model is already locked.
-        /// </summary>
-        public static CoreClrAssemblyLoader CreateAndSetDefault()
-        {
-            var assemblyLoader = new CoreClrAssemblyLoader();
-            return assemblyLoader;
-        }
+        private static readonly string[] _extensions = new[] { "ni.dll", "ni.exe", "dll", "exe" };
```

Comment:
No, this one is still needed. There was a separate list called `s_extensions` with *almost* the same information, and that one is no longer needed and has been removed.